velox/connectors/hive/HiveDataSink.h

```diff
@@ -238,6 +240,20 @@ class HiveInsertTableHandle : public ConnectorInsertTableHandle {
             "ensureFiles is not supported with partition keys in the data");
       }
     }
+
+    if (partitionKeyOrder_.size() > 0) {
+      folly::F14FastSet<std::string> partitionKeyNames(
+          partitionKeyOrder_.begin(), partitionKeyOrder_.end());
+      for (const auto& inputColumn : inputColumns_) {
+        if (inputColumn->isPartitionKey()) {
+          VELOX_CHECK(partitionKeyNames.erase(inputColumn->name()) == 1);
```

Comment:
I prefer sorting and comparing vs this deleting approach. But since the number of partition columns are often small, this should be okay.